internal/http/routes/api/v1/system.go

```diff
@@ -0,0 +1,74 @@
+package api_v1
+
+import (
+	"net/http"
+	"runtime"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-shiori/shiori/internal/dependencies"
+	"github.com/go-shiori/shiori/internal/http/context"
+	"github.com/go-shiori/shiori/internal/http/middleware"
+	"github.com/go-shiori/shiori/internal/http/response"
+	"github.com/go-shiori/shiori/internal/model"
+	"github.com/sirupsen/logrus"
+)
+
+type SystemAPIRoutes struct {
+	logger *logrus.Logger
+	deps   *dependencies.Dependencies
+}
+
+func (r *SystemAPIRoutes) Setup(g *gin.RouterGroup) model.Routes {
+	g.Use(middleware.AuthenticationRequired())
+	g.GET("/info", r.infoHandler)
+	return r
+}
+
+type infoResponse struct {
+	Version struct {
+		Tag    string `json:"tag"`
+		Commit string `json:"commit"`
+		Date   string `json:"date"`
+	} `json:"version"`
+	Database string `json:"database"`
+	OS       string `json:"os"`
+}
+
+// System info API endpoint godoc
+//
+//	@Summary					Get general system information
+//	@Description				Get general system information like Shiori version, database, and OS
+//	@Tags						system
+//	@Produce					json
+//	@securityDefinitions.apikey	ApiKeyAuth
+//	@Success					200	{object}	infoResponse
+//	@Failure					403	{object}	nil	"Only owners can access this endpoint"
+//	@Router						/api/v1/system/info [get]
+func (r *SystemAPIRoutes) infoHandler(c *gin.Context) {
+	ctx := context.NewContextFromGin(c)
+	if !ctx.GetAccount().Owner {
+		response.SendError(c, http.StatusForbidden, "Only owners can access this endpoint")
+		return
+	}
+
+	response.Send(c, 200, infoResponse{
+		Version: struct {
+			Tag    string `json:"tag"`
+			Commit string `json:"commit"`
+			Date   string `json:"date"`
+		}{
+			Tag:    model.BuildVersion,
+			Commit: model.BuildCommit,
+			Date:   model.BuildDate,
+		},
+		Database: r.deps.Database.DBx().DriverName(),
+		OS:       runtime.GOOS + " (" + runtime.GOARCH + ")",
```

Comment:
Can you specify how would you like to see this done? I already use a custom struct for the response, I initialise it directly in the `response.Send` to avoid setting it in memory temporarily only for one line.

Reply:
I think it's sightly better in terms of memory management, but since requests are ephemeral I don't see it as a huge deal.

Reply:
Yeah while I was working on this and on #825 I kept thinking about having code style documentation to make it clear for other people, the problem is that is a big amount of work and we don't get many contributors. But I could start a `go-shiori/code-style` repository and start putting things as I refactor things.